modules/core/src/main/java/org/apache/synapse/mediators/eip/splitter/CloneMediator.java

```diff
@@ -102,16 +103,32 @@ public boolean mediate(MessageContext synCtx) {
 
         synCtx.setProperty(id != null ? EIPConstants.EIP_SHARED_DATA_HOLDER + "." + id :
                            EIPConstants.EIP_SHARED_DATA_HOLDER, new SharedDataHolder());
+        //observability code starts here
+        String originalCorrelationId = null;
+        if (((Axis2MessageContext) synCtx).getAxis2MessageContext().
+                isPropertyTrue(PassThroughConstants.CORRELATION_LOG_STATE_PROPERTY)) {
+            originalCorrelationId = ((Axis2MessageContext) synCtx).getAxis2MessageContext().
+                    getProperty(PassThroughConstants.CORRELATION_ID).toString();
+        }
+        //observability code ends here
 
         // get the targets list, clone the message for the number of targets and then
         // mediate the cloned messages using the targets
         Iterator<Target> iter = targets.iterator();
         int i = 0;
         while (iter.hasNext()) {
             if (synLog.isTraceOrDebugEnabled()) {
-                synLog.traceOrDebug("Submitting " + (i+1) + " of " + targets.size() +
-                    " messages for " + (isSequential() ? "sequential processing" : "parallel processing"));
+                synLog.traceOrDebug("Submitting " + (i + 1) + " of " + targets.size() +
+                        " messages for " + (isSequential() ? "sequential processing" : "parallel processing"));
+            }
+
+            //Observability code
+            if (((Axis2MessageContext) synCtx).getAxis2MessageContext().
```

Comment:
we can simply check for originalCorrelationId is not null here, as check for correlation is enabled is already done above.